--- mtpaint-3.50/configure.orig	2025-10-07 20:02:50.525411967 +0200
+++ mtpaint-3.50/configure	2025-10-07 20:02:53.965432132 +0200
@@ -492,10 +492,6 @@ then
 # !!! Questionable, now; need retesting, and proper cross-configuring too
 	# With Clang, target pentium4; no incompatible HW expected in the wild,
 	# and anything lesser results in much slower code
-	case $ARCH in
-		i?86)	HAVE_CLANG && MARCH=${MARCH:--march=pentium4}
-			MARCH=${MARCH:--march=i386};;
-	esac
 	CFLAGS="-O2 $MARCH"
 	# Add optimizations which are proven to really make mtPaint code faster
 	if HAVE_CLANG
@@ -506,8 +502,6 @@ then
 	else # GCC 3.x
 		CFLAGS="$CFLAGS -fweb"
 	fi
-	# Do not add unneeded dependencies
-	AS_NEEDED="-Wl,--as-needed"
 	# Disable GTK+ debug code
 	DEFS="$DEFS -DGTK_NO_CHECK_CASTS -DG_DISABLE_CAST_CHECKS"
 elif [ "$OPTS" = DEBUG ]
@@ -523,7 +517,6 @@ else # $OPTS = CFLAGS
 	MARCH=
 	# And leave CFLAGS alone
 fi
-[ "$OPTS" != DEBUG ] && LDFLAGS="-s $LDFLAGS" # Strip debug info
 # Set Windows-specific flags
 if [ "$OS" != "${OS#MinGW/}" ]
 then
@@ -834,7 +827,8 @@ MT_PREFIX="$MT_PREFIX"
 MT_DATAROOT="$MT_DATAROOT"
 MT_LANG_DEST="$MT_LOCDIR"
 MT_MAN_DEST="$MT_MANDIR"
-LDFLAG = $AS_NEEDED $LIBS $LDFLAGS
+LDFLAG = $AS_NEEDED $LDFLAGS
+LIB = $LIBS
 CFLAG = $DEFS $INCLUDES $CPPFLAGS $CFLAGS
 subdirs = $MAKE_DIRS
 BIN_INSTALL="$MT_BINDIR"
--- mtpaint-3.50/src/Makefile.orig	2020-03-04 03:44:05.000000000 +0100
+++ mtpaint-3.50/src/Makefile	2025-10-07 19:59:11.350793850 +0200
@@ -3,6 +3,7 @@ include ../_conf.txt
 # Generate identical object files if possible
 CFLAGS = $(CFLAG) $(SET_RANDSEED)
 LDFLAGS = $(LDFLAG)
+LIBS = $(LIB)
 
 BIN = mtpaint$(EXEEXT)
 
@@ -12,7 +13,7 @@ OBJS = main.o mainwindow.o inifile.o png
 	cpick.o thread.o vcode.o
 
 $(BIN): $(OBJS)
-	$(CC) $(OBJS) -o $(BIN) $(LDFLAGS)
+	$(CC) $(CFLAG) $(LDFLAGS) $(OBJS) -o $(BIN) $(LIBS)
 
 $(OBJS): *.h graphics/*
 
